### PR TITLE
#164397755 Retrieve Intervention Records

### PR DESCRIPTION
**Pull request by ElishaMisoi**
**What does this PR do?**
Adds the Read in the CRUD for the intervention incidents

**Tasks to be completed**
- write tests for the get intervention endpoints
- add the get all records endpoint
- add the get one record endpoint

**How can this be manually tested**
Step 1

Navigate to your working directory, create and activate the virtual environment

```$ virtualenv -p python .venv```

```$ source .venv/bin/activate```

Clone the repository

```$ git clone https://github.com/ann-mukundi/ireporter.git```

Switch to the ireporter directory

```$ cd ireporter/```

Switch to testing branch

```$ git checkout ft-retrieve-interventions-164397755```

Install project dependencies

```$ pip install -r api/requirements.txt```

Run the database migrations

```$ python api/manage.py makemigrations```
```$ python api/manage.py migrate```

Step 2
**Running the application**

```$ python api/manage.py runserver```

Step 3
**Login user**
Signup and activate user (if no user exists)
```<your-localhost>/api/auth/signup```

Login user (after activation)
```<your-localhost>/api/auth/login```

Step 4
**Add interventions**
Use the token provided after login for Authorization and the endpoint
```<your-localhost>/api/interventions/```

Step 5
**Get all interventions**
Use the token provided after login for Authorization and the endpoint
```<your-localhost>/api/interventions/```

Step 6
**Get one interventions**
Use the token provided after login for Authorization and the endpoint
```<your-localhost>/api/interventions/<id>``` where ```id``` is the specific red-flag id

**OR**

**Get one rintervention**
Use the token provided after login for Authorization and the endpoint
```<your-localhost>/api/interventions/``` then click on the ```url``` in the one of the intervention details

**Relevent PT Stories**
[#164397755](https://www.pivotaltracker.com/story/show/164397755)

**Testing**

$ python manage.py test
**Prerequisites**

Python, Git, Virtualenv

**Screenshots**

**Get all interventions**
Screen Shot 2019-03-18 at 10 28 09

**Get one intervention**
Screen Shot 2019-03-18 at 10 28 54

**Intervention not found**
Screen Shot 2019-03-18 at 10 29 37

### CODE REVIEW

File: api/interventions/models.py [hunk 1]
```diff
@@ -8,7 +8,7 @@ class InterventionsModel(models.Model):
     The interventions model
     """
 
-    createdBy = models.IntegerField()
```
Comment (Benkimeric) [inline api/interventions/models.py]: This shoud be `    createdBy = models.IntegerField()
` and not the one you have accepted. CreatedBy should be an integer

Comment (Elisha-Misoi) [reply api/interventions/models.py]: Remember 'createdBy' should return a string of 'username' when retrieving. Setting it to an integer will not allow for this.

Comment (Benkimeric) [reply api/interventions/models.py]: When POSTing, it should return an integer, then when retrieving fetch the username based on the id field

Comment (Elisha-Misoi) [reply api/interventions/models.py]: It still returns the id of the user when posting an intervention and the username when retrieving the record. If we change the createdBy serializer field to an integer, during migrations the DB registers this as an integer hence no way of returning a string when retrieving.

**Review by AlvinMugambi — COMMENTED** [review]
You can add a `'url'` field in the intervention serializers so a URL to a specific intervention is displayed in the intervention details. 
Also the PR title requires changing.
Other than that all looks great!

**Comment by katherine95** [comment]
Format the PR message correctly. i.e `relevant PT story`. Use https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices for reference

**Comment by Siffersari** [comment]
The title 'What is the relevant PT story?' is not well formated. You could have it sound out as the rest of the titles. 
Good job

**Review by Siffersari — COMMENTED** [review]
Good job. I like that you are returning the username in the place of the user id. Also, returning null instead of an empty string.

**Comment by ElishaMisoi** [comment]
> You can add a `'url'` field in the intervention serializers so a URL to a specific intervention is displayed in the intervention details.
> Also the PR title requires changing.
> Other than that all look great!

Hey @AlvinMugambi I have made the change

**Comment by ElishaMisoi** [comment]
> Good job. I like that you are returning the username in the place of the user id. Also, returning null instead of an empty string.

Hey @Siffersari I have made the change, please check the screenshots

**Comment by ElishaMisoi** [comment]
> Format the PR message correctly. i.e `relevant PT story`. Use https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices for reference

Hey @katherine95 I have made the change

**Review by AlvinMugambi — APPROVED** [review]

**Pull request merged**